tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,206 @@ void test_bsearch(void)
 	zassert_not_null(result, "bsearch -key found");
 }
 
+/**
+ *
+ * @brief Test abs function
+ *
+ */
+
+void test_abs(void)
+{
+	int val = -5, value = 5;
+
+	zassert_equal(abs(val), 5, "abs -5");
+	zassert_equal(abs(value), 5, "abs 5");
+}
+
+/**
+ *
+ * @brief Test atoi function
+ *
+ */
+
+void test_atoi(void)
+{
+	char *str[6] = {"123", "2c5", "acd", "6 7a", "3-4e", "8+1c"};
+
+	zassert_equal(atoi(str[0]), 123, "atoi error");
+	zassert_equal(atoi(str[1]), 2, "atoi error");
+	zassert_equal(atoi(str[2]), 0, "atoi error");
+	zassert_equal(atoi(str[3]), 6, "atoi error");
+	zassert_equal(atoi(str[4]), 3, "atoi error");
+	zassert_equal(atoi(str[5]), 8, "atoi error");
+}
+
+/**
+ *
+ * @brief Test value type
+ *
+ * @details This function check the char type,
+ * and verify the return value.
+ *
+ * @see isalnum(), isalpha(), isdigit(), isgraph(),
+ * isprint(), isspace(), isupper(), isxdigit().
+ *
+ */
+void test_checktype(void)
+{
+	int i;
```

Comment:
That's really not what I was suggesting.  I meant something like this:
```
  char buf[128];
  const char exp_alnum[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
  const char exp_isspace[] = "\t\n\v\f\r";
  ...
  char *bp = buf;

  for (int i = 0; i < 128; i++) {
    if (isalnum(i)) {
      *bp++ = i;
    }
  }
  *bp = 0;
  zassert_equal(strcmp(buf, exp_alnum));

  bp = buf;
  for (int i = 0; i < 128; i++) {
    if (isspace(i)) {
      *bp++ = i;
    }
  }
  *bp = 0;
  zassert_equal(strcmp(buf, exp_space));

  ....
```
which somebody looking at can be convinced tests things correctly.  It's not obvious what `((unsigned)i|32u)-(unsigned)'a') < 26U)` has to do with checking whether a character is alphabetic.

Reply:
OK, I got your point.